rasa_nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -0,0 +1,513 @@
+        self.intent_split_symbol = params_dict.get("intent_split_symbol",
+                                                   self.intent_split_symbol)
+
+    def train(self, training_data, config, **kwargs):
+        # type: (TrainingData, RasaNLUModelConfig, **Any) -> None
+        """Train the embedding intent classifier on a data set."""
+
+        self._extract_params_from_config(config)
+
+        # transform intents to numbers
+        # if intent_tokenization = False these dicts are identical
+        self.intent_dict, self.intent_token_dict = self._create_intent_dicts(
+                training_data)
+
+        # check if number of negatives is less than number of intents
+        logger.debug("Check if num_neg {} is smaller than number of intents {}, "
+                     "else set num_neg to the number of intents".format(
+                      self.num_neg, len(self.intent_dict)))
+        self.num_neg = min(self.num_neg, len(self.intent_dict) - 1)
+
+        # create an array for mu
+        self.mu = self.mu_neg * np.ones(self.num_neg + 1)
+        self.mu[0] = self.mu_pos
+
+        X = np.stack([example.get("text_features")
+                      for example in training_data.intent_examples])
+
+        # due to tokenization of the intents, special array is created
+        # that stores the number of the whole intent from intent_dict
+        intents_for_X = np.zeros(X.shape[0], dtype=int)
+        for i, example in enumerate(training_data.intent_examples):
+            intents_for_X[i] = self.intent_dict[example.get("intent")]
+
+        # array that holds bag of words for tokenized intents
+        # if intent_tokenization = False this is one-hot vector
+        Y = np.zeros([X.shape[0], len(self.intent_token_dict)])
+        for i, example in enumerate(training_data.intent_examples):
+            if self.intent_tokenization_flag and self.intent_split_symbol:
+                for t in example.get("intent").split(self.intent_split_symbol):
+                    Y[i, self.intent_token_dict[t]] = 1
+            else:
+                Y[i, self.intent_dict[example.get("intent")]] = 1
+
+        # the matrix that encodes intents as bag of words in rows
+        # if intent_tokenization = False this is identity matrix
+        encoded_all_intents = self._create_encoded_intents()
+
+        # stack encoded_all_intents on top of each other
+        # to create candidates for training examples
+        # to calculate training accuracy
+        all_Y = np.stack([encoded_all_intents for _ in range(X.shape[0])])
+
+        with self.graph.as_default():
+            a_in = tf.placeholder(tf.float32, (None, X.shape[-1]), name='a')
+            b_in = tf.placeholder(tf.float32, (None, None, Y.shape[-1]),
+                                  name='b')
+
+            self.embedding_placeholder = a_in
+            self.intent_placeholder = b_in
+
+            is_training = tf.placeholder(tf.bool)
```

Comment:
I believe building this as `tf.placeholder_with_default` might make it easier - the default could be false and therefore we wouldn't need to save it for processing messages

Reply:
I'll change that